JavaScriptFile.js

```diff
@@ -33,10 +42,36 @@ var JavaScriptFile = function(props) {
     this.project = props.project;
     this.pathName = props.pathName;
     this.type = props.type;
-    this.API = props.project.getAPI();
+    this.API = this.project.getAPI();
 
     this.logger = this.API.getLogger("loctool.plugin.JavaScriptFile");
     this.set = this.API.newTranslationSet(this.project ? this.project.sourceLocale : "zxx-XX");
+    this.mapping = this.type.getMapping(this.pathName);
+
+    this.localeSpec = props.locale || (this.mapping && this.API.utils.getLocaleFromPath(this.mapping.template, this.pathName)) || "en-US";
+    this.locale = new Locale(this.localeSpec);
+
+    // get the regexp that finds the function call that wraps strings to translate
+    var jsSettings = this.project.settings && this.project.settings.javascript;
+    var wrapper = (jsSettings && jsSettings.wrapper) || "(^R|\\WR)B\\s*\\.\\s*getString(JS)?";
```

Comment:
I think it needs an additional test case for that case different regular expression than the default one, and make sure if it works ok. what do you think?
What if the user wants the default one and an additional case? for example, webos-javascript plugins parse both getString and $L

Reply:
I did that sort of test in the loctool samples, but we can do it here as a unit test as well.

Reply:
For both of them, you would use alternation:

```json
{
    "wrapper": "(\\.\\s*getString(JS)?|\\$L)"
}
```

Reply:
Actually, there are already unit tests for it at the end of testJavaScriptFile.js: `testJavaScriptFileParseWithAlternateWrapper` and `testJavaScriptFileParseWithAlternateWrapperAndEmbeddedQuote`

Reply:
oh I see.